pipeline/build/build_test.go

```diff
@@ -193,26 +208,28 @@ func TestRunInvalidLdflags(t *testing.T) {
 func TestRunPipeFailingHooks(t *testing.T) {
 	assert := assert.New(t)
 	var config = config.Project{
-		Build: config.Build{
-			Hooks: config.Hooks{},
-			Goos: []string{
-				runtime.GOOS,
-			},
-			Goarch: []string{
-				runtime.GOARCH,
+		Builds: []config.Build{
+			{
+				Hooks: config.Hooks{},
+				Goos: []string{
+					runtime.GOOS,
+				},
+				Goarch: []string{
+					runtime.GOARCH,
+				},
 			},
 		},
 	}
 	var ctx = &context.Context{
-		Config:   config,
-		Archives: map[string]string{},
+		Config:  config,
+		Folders: map[string]string{},
 	}
 	t.Run("pre-hook", func(t *testing.T) {
-		ctx.Config.Build.Hooks.Pre = "exit 1"
+		ctx.Config.Builds[0].Hooks.Pre = "exit 1"
 		assert.Error(Pipe{}.Run(ctx))
 	})
 	t.Run("post-hook", func(t *testing.T) {
-		ctx.Config.Build.Hooks.Post = "exit 1"
+		ctx.Config.Builds[0].Hooks.Post = "exit 1"
 		assert.Error(Pipe{}.Run(ctx))
```

Comment:
actually, this is being tested already!

https://github.com/goreleaser/goreleaser/blob/master/pipeline/defaults/defaults_test.go#L75

the single build will always become an item of the builds list :)